### ISSUE TITLE
Create opening database to ensure diverse training

### ISSUE THREAD

**Issue by dtracers** [LeelaChessZero/lc0#733]
Temperature while creating a variety of openings does not ensure that openings that leela does not understand are explored and instead is more based on luck.

Instead I suggest that we use the data created for the kdlgain threshold to force openings.
ref: #721 #681 

We can compute how well leela understands a position after the fact:
[image of eval for game 58 of tcec that leela lost](https://media.discordapp.net/attachments/430695662108278784/546045833934798850/Screen_Shot_2019-02-15_at_11.07.40_AM.png?width=800&height=390)

Using the knowledge of how well leela understood a position over vast majority of games we should be able to find open sequences that lead to positions that leela does not understand.
With the sequences we can then force them during training to increase knowledge of those positions and openings.
 
This should also allow us to detect information loss / overfitting is occurring. 
But should also help prevent information loss as well as it should play openings that leela is starting to forget.

Steps needed to make this happen:

- Need a way to truly quantify leela understanding of a game it played.
- The server needs a way to track openings played.
- The client needs a way for the server to force openings or (books) on it.

**Comment by Naphthalin** [comment]
As far as I can tell, all raised issues are resolved now:
- opening diversity is acceptable now thanks to `--policy-softmax-temp=1.2` in training
- openings played are tracked via external analysis (see !openings, !openings2 and the occasional Chad analysis in Discord)
- opening books are possible in training and used e.g. for FRC in T71, and also back in T49.

This issue can be probably be closed.

**Comment by mooskagh** [comment]
I still have an idea to have a database of positions for Lc0 (blunders/openings/fortresses) for various purposes (mostly, regression testing).

But I'm not sure whether keeping the issue open helps in any way..
